### PR TITLE
feat(cli): Adding console for get & list methods

### PR DESCRIPTION
**Pull request by byawitz**
## What does this PR do?
Added `--console` flag to open resources at project console.

## Test Plan

image

### CODE REVIEW

File: templates/cli/lib/utils.js.twig [hunk 1]
```diff
@@ -14,6 +18,60 @@ function getAllFiles(folder) {
...
+
+    const end = action === 'get' ? (id ? `/${resource}-${id}` : `/${resource}`) : '';
+    const projectId = localConfig.getProject().projectId;
+    const middle = resource !== '' ? `/project-${projectId}` : '';
+    const url = `${baseUrl}/console${middle}/${service}${end}`
+
+    const start = (process.platform == 'darwin' ? 'open' : process.platform == 'win32' ? 'start' : 'xdg-open');
+    const key = (process.platform == 'darwin' ? 'Return' : 'Enter');
+
+    success(`\n  ${url}\n  Press <${key}> to open URL in your default browser, exising in 3 seconds`);
```
Comment (Meldiron) [inline templates/cli/lib/utils.js.twig:64]: I would exit after 5 secs instead. 3 is too fast to read the sentence.

File: templates/cli/lib/utils.js.twig [hunk 2]
```diff
@@ -14,6 +18,60 @@ function getAllFiles(folder) {
...
+    setTimeout(() => process.exit(0), 3000);
```
Comment (Meldiron) [inline templates/cli/lib/utils.js.twig:65]: Not sure if I like the timer. I would probably just show URL, and give hint they can use tag `--open` to open it in browser automatically.

File: templates/cli/base/requests/api.twig [hunk 3]
```diff
@@ -17,8 +17,19 @@
     fs.writeFileSync(destination, response);
     {%~ endif %}
     if (parseOutput) {
+    {%~ if method.name == 'get' and service.name not in ['health','migrations','locale'] %}
+        if(console) {
+            showConsoleLink('{{service.name}}', 'get'
```
Comment (Meldiron) [inline templates/cli/base/requests/api.twig:22]: Does this support stuff like `appwrite functions getDeployment`?

File: templates/cli/base/requests/api.twig [hunk 4]
```diff
@@ -17,8 +17,19 @@
     fs.writeFileSync(destination, response);
     {%~ endif %}
     if (parseOutput) {
+    {%~ if method.name == 'get' and service.name not in ['health','migrations','locale'] %}
```
Comment (Meldiron) [inline templates/cli/base/requests/api.twig:20]: Can we make this supoprt list too? `appwrite functions list` or `appwrite functions listExecutions`

File: templates/cli/lib/utils.js.twig [hunk 5]
```diff
@@ -14,6 +18,56 @@ function getAllFiles(folder) {
...
+    const end = action === 'get' ? (id ? `/${resource}-${id}` : `/${resource}`) : '';
+    const projectId = localConfig.getProject().projectId;
+    const middle = resource !== '' ? `/project-${projectId}` : '';
+    const url = `${baseUrl}/console${middle}/${service}${end}`
+
+
+    success(url);
+
+    if (open) {
+        const start = (process.platform == 'darwin' ? 'open' : process.platform == 'win32' ? 'start' : 'xdg-open');
```
Comment (PineappleIOnic) [inline templates/cli/lib/utils.js.twig:65]: Do we have a fallback for if this fails? Maybe on a entirely CLI environment like a server or CI we could display a "If your browser does not automatically open please copy the link 'xxxxxxxxx'"

**Review by PineappleIOnic — APPROVED** [review]

**Comment by byawitz** [comment]
Expanding console flow support 

https://github.com/appwrite/sdk-generator/assets/316103/89c6ae8b-7d3a-46e5-af66-103e8c9e212d

File: templates/cli/lib/commands/command.js.twig [hunk 6]
```diff
@@ -94,6 +95,10 @@ const {{ service.name | caseLower }}{{ method.name | caseUcfirst }} = async ({
 {% if method.type == 'location' %}
     .requiredOption(`--destination <path>`, `output file path.`)
 {% endif %}
+{% if methodHaveConsolePreview(method.name,service.name) %}
+    .option(`--console`, `Get the resource console url`)
+    .option(`--open`, `Use with '--console' to open the using default browser`)
```
Comment (christyjacob4) [inline templates/cli/lib/commands/command.js.twig:100]: shouldn't this be removed?

Comment (byawitz) [reply templates/cli/lib/commands/command.js.twig:100]: Yes, when we will merge the one after

File: src/SDK/Language/CLI.php [hunk 7]
```diff
@@ -290,4 +348,18 @@ public function getParamExample(array $param): string
...
+
+    /**
+     * Language specific filters.
+     * @return array
+     */
+    public function getFunctions(): array
+    {
+        return [
+            /** Return true if the entered service->method is enabled for a console preview link */
+            new TwigFunction('methodHaveConsolePreview', fn($method, $service) => preg_match('/^([Gg]et|[Ll]ist)/', $method)
```
Comment (christyjacob4) [inline src/SDK/Language/CLI.php:360]:
```suggestion
            new TwigFunction('hasConsolePreview', fn($method, $service) => preg_match('/^([Gg]et|[Ll]ist)/', $method)
```

File: templates/cli/base/requests/api.twig [hunk 8]
```diff
@@ -17,8 +17,19 @@
     fs.writeFileSync(destination, response);
     {%~ endif %}
     if (parseOutput) {
+    {%~ if methodHaveConsolePreview(method.name,service.name) %}
```
Comment (christyjacob4) [inline templates/cli/base/requests/api.twig:20]:
```suggestion
    {%~ if hasConsolePreview(method.name,service.name) %}
```

File: templates/cli/lib/commands/generic.js.twig [hunk 9]
```diff
@@ -122,8 +151,51 @@ const logout = new Command("logout")
...
+
+            return;
+        }
+
+        const answers = await inquirer.prompt(questionsLogout);
+
+        if (answers.accounts) {
+            for (let accountId of answers.accounts) {
+                globalConfig.setCurrentSession(accountId);
+                await deleteSession(accountId);
```
Comment (christyjacob4) [inline templates/cli/lib/commands/generic.js.twig:185]: we are setting the current session to an accountID and deleting that specific one? Which means all new requests will throw 401 ?

File: templates/cli/lib/commands/generic.js.twig [hunk 10]
```diff
@@ -122,8 +151,51 @@ const logout = new Command("logout")
...
+            }
+        }
+
+        const leftSessions = globalConfig.getSessions();
```
Comment (christyjacob4) [inline templates/cli/lib/commands/generic.js.twig:189]:
```suggestion
        const remainingSessions = globalConfig.getSessions();
```
Lets use this name or something more simple if you got ideas

File: templates/cli/lib/commands/generic.js.twig (hunk 9 above)
Comment (byawitz) [reply templates/cli/lib/commands/generic.js.twig:185]: We're setting the current session to the last account in line using the `leftSession` variable right after.

File: templates/cli/lib/utils.js.twig [hunk 11]
```diff
@@ -1,9 +1,13 @@
 const fs = require("fs");
 const path = require("path");
+const { localConfig, globalConfig } = require("./config");
+const { success, log, error } = require('./parser')
+const readline = require('readline');
+const cp = require('child_process');
```
Comment (christyjacob4) [inline templates/cli/lib/utils.js.twig:6]: lets remove this from all files now that we are not opening stuff

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**